src/styles/main.scss

```diff
@@ -1,3 +1,81 @@
+$days: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6
+);
+
+$body-color: #000;
+
 body {
   margin: 0;
+  color: $body-color;
+}
+
+.container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  height: 100vh;
+}
+
+.calendar {
+  box-sizing: border-box;
+  gap: 1px;
+  display: grid;
+  grid-template-columns: repeat(7, 1fr);
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  align-items: flex-start;
+  padding: 10px;
+
+  .calendar__day {
+    box-sizing: border-box;
+    border: 1px solid #000;
+    background-color: #aaa;
+    transition: transform 500ms;
+    cursor: pointer;
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: #faa;
+    }
+
+    @for $length from 28 through 31 {
+      .calendar.length-#{$length} {
+        grid-column-end: span #{$length};
+      }
+    }
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day}) {
+        &::before {
+          content: "#{$day}";
+          display: flex;
+          justify-content: center;
+          align-items: center;
+          height: 100px;
+          width: 100px;
+        }
+      }
+    }
+
+    @for $length from 28 through 31 {
```

Comment:
there is a mess in selectors, loops, and CSS properties
implement every modificatior one by one and only write the code exactly needed for it and you know what it do